### DISCUSSION TITLE
How can I enable ohmyposh only for the PowerShell in vsCode envirment without needing the PowerShell vsCode extension

### DISCUSSION THREAD

**Discussion by babakfp** [JanDeDobbeleer/oh-my-posh#4868] [Troubleshoot] [answered]
Hi 👋

I want to use ohmyposh only in the PowerShell that is open in vsCode. This is possible by installing the PowerShell extension and using the `Microsoft.VSCode_profile.ps1` file. But, I don't want to use the extension. Is there a way to achieve this?

I found out about `.Env.TERM_PROGRAM` which returns `vscode` in vsCode PowerShell. But, I don't want this either.

**Comment by JanDeDobbeleer** [comment]
@babakfp I don't understand the question. Vscode can use any shell, which I think also includes your default pwsh instance.

**Reply by babakfp** [reply]
I can open PowerShell inside vsCode or by itself.

When I open it by itself, I don't want it to be using ohmyposh, but when I open it inside vsCode, I want it to use it.

By using PowerShell by itself I mean opening the PowerShell like this:

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/44144724/dacb21ec-6d1c-4abe-bef5-c48e315488e3)

And this is how I'm using it in vsCode:

![image](https://github.com/JanDeDobbeleer/oh-my-posh/assets/44144724/cbc60c66-8eb6-4365-abf4-da550b1a5657)

This vsCode extension https://marketplace.visualstudio.com/items?itemName=ms-vscode.PowerShell allows using a separate PowerShell profile file for the PowerShell that you open inside vsCode. I don't want to use it.

I may be asking this in the wrong place.

**Reply by JanDeDobbeleer** [reply]
@babakfp assuming that they bot use the same profile, you indeed need to add a switch inside it that only initializes oh-my-posh when running inside VSCODE. So, you'll need to use `if -not $Env.TERM_PROGRAM -eq 'vscode'`.

**Reply by babakfp** [reply, marked as answer]
Ahhh, right, how couldn't I think about this!

Result `Microsoft.PowerShell_profile.ps1`:

```powershell
if ($Env:TERM_PROGRAM -eq "vscode") {
    oh-my-posh init pwsh -c "C:\Users\Babak\Documents\OhMyPoshThemes\main.json" | Invoke-Expression
}
```

Thanks 🌸

---

Can I open a PR and add this to the docs (if it doesn't already exist 🤦‍♂️)?

**Reply by JanDeDobbeleer** [reply]
@babakfp this question actually had no place in our docs as it's not related to oh-my-posh.
